vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp

```diff
@@ -1,23 +1,205 @@
+        $("#description").hide()
+        $("#newDescription").val(description)
+        $("#newDescription").data('value',description);
+        $("#editDescription").hide()
+        $("#submitDescription").show()
+        $("#cancelEditDescription").show()
+        
+        
+    });
+
+    
+    // --- Submit Description ------
+    
+    $("#submitDescription").click(function() {
+        var formData = new FormData();
+        formData.append('description', $("#newDescription").val());
+        $.ajax({
+        url: "<c:url value="/staff/module/${module.id}/sequence/${sequence.id}/edit/description?${_csrf.parameterName}=${_csrf.token}" />",
+        type: 'POST',
+        cache       : false,
+        contentType : false,
+        processData : false,
+        data: formData,
+        enctype: 'multipart/form-data',
+        success: function(data) {
+            // replace text box with new description
+            $("#submitDescription").hide()
+            $("#cancelEditDescription").hide()
+            $("#editDescription").show()
+            var val = $("#newDescription").val();
+            $("#description").text(val);
+            $("#editDescription").show()
+            $("#description").show() 
+            $('#description').insertBefore( "#newDescription" );
+            $("#description").data('value',val);
+            $("#newDescription").remove()
+        },
+        error: function(data) {
+            $(".open").removeClass("open");
+                var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                $('.error').append(alert);
+            }
+        });
+        
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
+        $('description').insertBefore( "#newDescription" );
+        $("#newDescription").remove()
+        $("#description").show()
+        
+    });
+    
+    // ------- Edit/Save/Cancel Description ends --------
+    
+    
+    
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    $("#editTitle").click(function() {
+    	// Using data attribute
+    	var sequenceTitle = $("#titleName").data('value')  // gets value without using trim function
+        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	$("#title").val(sequenceTitle)
+    	$('#title').text('Sequence: ')
+        $("#newTitle").val(sequenceTitle)
```

Comment:
why is there a title and a newtitle? what's that for? And since they are h2 and p, why do you use val() here?

Reply:
Since title is <h2> tag, while editing title, newTitle of the input form is added so that user can add any new title.

Reply:
Yes, I will remove the val()

Reply:
I'm still confused why you don't use the h2 tag?

Reply:
Because, in the code,
When editTitle function is called, it sets the sequence value to "Sequence:", due to which I cannot retrieve the value from the h2 tag.

$("#editTitle").click(function() {
// Using data attribute
var sequenceTitle = $("#title").data('value') // gets value without using trim function
$('

').insertAfter( "#title" );
$("#title").val(sequenceTitle)
**$('#title').text('Sequence: ')**


Hence, I have added p tag to get the value back.

Reply:
I can only retrieve "Sequence :" as value, and not the actual value, e.g. **Sequence value: abcd**

Reply:
I don't get it, is it a display issue? Why don't you do
```
$('#title').data('previous-title', sequenceTitle);
```
when start editing and then
```
var sequenceTitle = $("#title").data('previous-title')  // gets value
$("#title").text("Sequence: "+sequenceTitle)
```
?

Reply:
I can't use data attributes in title, since it's an H2 tag right?

Reply:
Hence, I am using a p tag with data attributes

Reply:
https://www.w3schools.com/tags/att_data-.asp

Reply:
Okay, I will use data-attributes in h2 tag and remove the p tag.